### PR TITLE
Handlers to functions with receiver

### PR DESCRIPTION
**Pull request by vjgarciag96**
This is finally ready.

I have migrated all the handlers available in the library to now use functions with receivers. In this way it'll be possible to access to all the data correspondent to the updates in the scope of the handler function without the need of explicitly indicating these data parameters in the handler lambda function.

There is one commit per migrated handler (all the commits that reference the issue https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/issues/96)

Also, there are some clean up commits:

* https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/pull/103/commits/e1978f8e0a343323878028a6bea516a582311fb2 
* https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/pull/103/commits/8d7d45df9b2760b8de799f291efe3cf20db1b069
* https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/pull/103/commits/c9a0b5f6e5ec34db130863ba620c11072270b702

If these changes look good to you, we can finally proceed with this issue https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/issues/93

### CODE REVIEW

**Comment by y9san9** [comment]
wow nice, waiting for merge!

File: samples/dispatcher/src/main/kotlin/com/github/kotlintelegrambot/dispatcher/Main.kt
Comment (seik) [inline samples/dispatcher/src/main/kotlin/com/github/kotlintelegrambot/dispatcher/Main.kt:36]: I love that we don't have to use the `!!` operator anymore

**Review by seik — APPROVED** [review]
Huge PR.

I tried to read all very carefully but I feel I probably missed something. I'm waiting on this one for @manolovn to also approve since I don't trust myself.

**Comment by seik** [comment]
Moving here the conversation since the other PR is closed. @vjgarciag96 what's your view on #108 ?

**Comment by seik** [comment]
Adding this: https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/pull/108#issuecomment-703253323

**Comment by vjgarciag96** [comment]
Will you be able to take a look to the PR @manolovn? If not, we can try to find another reviewer

**Comment by reopold** [comment]
Can I report some problem in this PR?
Under Windows:
``` 
176 tests completed, 3 failed
> Task :telegram:test FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':telegram:test'.
```
The following tests fail::
`sendMediaGroup with media group composed by a video(fileUrl), a photo(fileId) and all the arguments`
`sendMediaGroup with media group composed by a video(fileUrl), a photo(fileId) and only the mandatory arguments`
`setWebhook with certificate as file`

The problem is in \r\n. 
**requestBody** that is constructed by the system contains only **\n** 
**expectedRequestBody** that is loaded from the resource file contains **\r\n**

req

**Comment by vjgarciag96** [comment]
It looks like @manolovn is not going to be able to review it

Is anyone interested on contributing by reviewing and giving feedback in this PR? @reopold @aNNiMON @SavinMike @red-avtovo @imorte

**Comment by imorte** [comment]
> It looks like @manolovn is not going to be able to review it
> 
> Is anyone interested on contributing by reviewing and giving feedback in this PR? @reopold @aNNiMON @SavinMike @red-avtovo @imorte

okay, add me to reviewers

**Comment by red-avtovo** [comment]
Let me take a look on it

**Comment by vjgarciag96** [comment]
Thanks @imorte and @red-avtovo! 

I sent you an invite to add you as collaborators to the project to be able to add you as reviewers in this PR 🙂

**Review by manolovn — APPROVED** [review]

**Review by imorte — APPROVED** [review]

**Review by imorte — APPROVED** [review]

**Review by red-avtovo — APPROVED** [review]

**Comment by vjgarciag96** [comment]
Ok, so now we have 4 approvals, thank you so much for your help folks! I'm going to merge this 💪🏻

**Comment by imorte** [comment]
could you please create a release? 🙂

**Comment by vjgarciag96** [comment]
Sure @imorte! I wanted to take a look to another thing before releasing 6.0.0 (more info here https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/issues/93), but the release should be ready by this Sunday.

**Pull request merged**
